### PR TITLE
touch-mode option added (closes #984)

### LINKED ISSUE #984: TestCafe should not emulate touch events for touch monitors
### Are you requesting a feature or reporting a bug?

feature

### What is the current behavior?

TestCafe switches in touch mode not only for testing in mobiles and tablets but in laptops and desktops with a monitor that supports touch events. But usually you expect to see mouse behavior when testing on a desktop. 

### What is the expected behavior?

I propose to switch in touch mode in desktops only if the `--touch-mode` flag enabled.

### How would you reproduce the current behavior (if this is a bug)?
The problem is reproduced on the http://demos.devexpress.com/bootstrap/GridView/Overview.aspx page. 
On the original page you can drag a column header with mouse but can't do it via touch events. So it's impossible to test this functionality on desktop with touch-monitor because TestCafe runs tests only with touch mode in this case.

/cc @inikulin @helen-dikareva

### PR DESCRIPTION
**Pull request by helen-dikareva**

### CODE REVIEW

**Comment by helen-dikareva** [comment]
Wait for https://github.com/DevExpress/testcafe-hammerhead/pull/990
and
https://github.com/DevExpress/testcafe-legacy-api/pull/24

File: src/client/core/utils/event.js [hunk 1]
```diff
@@ -71,3 +74,7 @@ export function documentReady () {
         }
     });
 }
+
+export function setTouchMode (mode) {
```
Comment (AlexanderMoskovkin) [inline src/client/core/utils/event.js]: maybe `setTouchMode (enabled)` ?

File: src/client/ui/select-element.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ function clickOnOption (optionIndex, isOptionDisabled) {
     var curSelectIndex   = curSelectEl.selectedIndex;
     var realOption       = curSelectEl.getElementsByTagName('option')[optionIndex];
     var clickLeadChanges = !isOptionDisabled && optionIndex !== curSelectIndex;
-    var isMobileBrowser  = browserUtils.isSafari && browserUtils.hasTouchEvents || browserUtils.isAndroid;
+    var isMobileBrowser  = browserUtils.isSafari && eventUtils.touchMode || browserUtils.isAndroid;
```
Comment (AlexanderMoskovkin) [inline src/client/ui/select-element.js]: Lets replace `browserUtils.isSafari && eventUtils.touchMode || browserUtils.isAndroid;` with a single flag from browserUtils

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by churkin** [comment]
@testcafe-build-bot \retest

**Comment by churkin** [comment]
@testcafe-build-bot \retest

File: test/functional/fixtures/is-touch-device.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    var isIOS       = /(iphone|ipod|ipad)/.test(userAgent);
+    var isAndroid   = /(android)/.test(userAgent);
+
+    var isDevice       = mobile || tablet || nexusDevice || isIOS || isAndroid || blackberry;
+    var hasTouchEvents = 'ontouchstart' in window;
+
+
+    return isDevice && hasTouchEvents;
+}
+/*eslint-disable no-unused-vars*/
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/is-touch-device.js]: I think it's better don't mix tests with util files. So, I'd like to move out this file from the `fixtures` directory

**Pull request merged**